### PR TITLE
txn_types: Check overflow in timestamp computation

### LINKED ISSUE #9771: Check overflow in timestamp computation
## Development Task
[`TimesStamp`](https://github.com/tikv/tikv/blob/master/components/txn_types/src/timestamp.rs#L8) is an u64 integer. There are some places where TiKV increases or decreases it but TiKV doesn't check overflow. Overflow can result in an infinite loop. Even if it's not possible to happen in theory, it's better to check it.

### PR DESCRIPTION
**Pull request by TszKitLo40**
Signed-off-by: Zijie Lu <wslzj40@gmail.com>

### What problem does this PR solve?

Issue Number: close #9771 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
Check overflow in timestamp computation
### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 
- No release note

### CODE REVIEW

File: components/txn_types/src/timestamp.rs [hunk 1]
```diff
@@ -33,20 +33,24 @@ impl TimeStamp {
     }
 
     pub fn next(self) -> TimeStamp {
+        assert!(self.0 + 1 >= 0);
```
Comment (andylokandy) [inline components/txn_types/src/timestamp.rs:36]: Maybe we need this:
```suggestion
        assert!(self.0 < u64::max());
```

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by andylokandy — APPROVED** [review]

**Pull request merged**